indigo_api/migrations/0019_backfill_principal_works.py

```diff
@@ -0,0 +1,33 @@
+# Generated by Django 3.2.12 on 2022-11-03 09:26
+import logging
+
+from django.db import migrations
+
+
+log = logging.getLogger(__name__)
+
+
+def backfill_principal_works(apps, schema_editor):
+    """ Mark existing works as principal works if the works:
+    - have at least one document associated with them; OR
+    - works that do not repeal, amend or commence another work
+    """
+    Work = apps.get_model('indigo_api', 'Work')
+    db_alias = schema_editor.connection.alias
+
+    for work in Work.objects.using(db_alias).iterator(100):
+        if work.document_set.exists() or not work.parent_work:
```

Comment:
Ignore `parent_work`. Instead, check for `work.commencements_made`, `work.amendments_made` and `work.repealed_works`